Challenge/View/Movies/MoviesList/MoviesListView.swift

```diff
@@ -0,0 +1,259 @@
+//
+//  MoviesListView.swift
+//  Challenge
+//
+//  Created by Lucas Henrique Machado da Silva on 06/05/2018.
+//  Copyright © 2018 Lucas Henrique Machado da Silva. All rights reserved.
+//
+
+import UIKit
+
+class MoviesListView: UIViewController {
+    // MARK: - Outlets
+    @IBOutlet weak var tvMovies: UITableView!
+    @IBOutlet weak var aivLoader: UIActivityIndicatorView!
+    @IBOutlet weak var lMessage: UILabel!
+    @IBOutlet weak var lMore: UILabel!
+    @IBOutlet weak var sbName: UISearchBar!
+    
+    // MARK: - Constants
+    private let refreshControl = UIRefreshControl()
+    
+    // MARK: - Presenter
+    var presenter: MoviesListPresenter?
+    
+    // MARK: - Variables
+    var movies: [Movie]?
+    var selectedMovie: Movie?
+    var loadingMoreMovies: Bool = false
+    var allPagesLoaded: Bool = false
+    
+    // MARK: - View Lifecycle
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        self.presenter = MoviesListPresenter(self)
+        
+        self.setupView()
+    }
+    
+    // MARK: - Setups
+    func setupView() {
+        self.aivLoader.startAnimating()
+        
+        self.tvMovies.accessibilityIdentifier = "tvMovies"
+        self.tvMovies.allowsSelection = true
+        if #available(iOS 10.0, *) {
+            self.tvMovies.refreshControl = self.refreshControl
+        } else {
+            self.tvMovies.insertSubview(refreshControl, at: 0)
+        }
+        self.refreshControl.tintColor = UIColor(red: 145 / 255, green: 142 / 255, blue: 244 / 255, alpha: 1.0)
+        self.refreshControl.addTarget(self, action: #selector(self.refreshTableView), for: .valueChanged)
+        
+        self.presenter?.getUpcomingMovies()
+    }
+    
+    // MARK: - Navigation
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
```

Comment:
Why do you use segues here ?  Do you see any other options ?

Reply:
@arctouch-ruannhomem I think when calling another view in storyboard (and passing data throw it) the easiest way is to use segues, I could also manually load a view from the storyboard and present it, but I thing it is kind of useless in this case. If it was a XIB based project I would need to do it all manually with present / push view controller.